### PR TITLE
SW-1077 Increase Laserhead max. working Temperature

### PR DESCRIPTION
**Pull request by ahmed-mrbeam**
-- Created new folder profiles-->laserheads which contains laserheads yaml properties
-- Extended iobeam-->laserhead_handler to procide a new property laserhead max temprature
-- Added necessary changes to temperature manager
-- in mrbeam plugin init file, reversed the sequence of the init between temprature manager and laserhead handler for dependency reasons
-- Marked the max laserhead temperature found in default printer profile as deprecated

### CODE REVIEW

File: octoprint_mrbeam/iobeam/laserhead_handler.py [hunk 1]
```diff
@@ -407,3 +409,65 @@ def _write_laser_heads_file(self, laser_heads_file=None):
                 self._logger.info("Writing to file: {} ..is successful!".format(laser_heads_file))
         except IOError as e:
             self._logger.error("Can't open file: {} , Due to error: {}: ".format(laser_heads_file, e))
+
+    @property
+    def current_laserhead_max_temperature(self):
+        """
+        Return the current laser head max temperature
+        Returns:
+        Laser head max temp (float)
```
Comment (JFSolution) [inline octoprint_mrbeam/iobeam/laserhead_handler.py]: There is a space needed between the description and the returns otherwise it will show wrong in the IDE and the returns have to be tabed once

File: octoprint_mrbeam/iobeam/laserhead_handler.py [hunk 2]
```diff
@@ -407,3 +409,65 @@ def _write_laser_heads_file(self, laser_heads_file=None):
...
+            return LASERHEAD_MAX_TEMP_FALLBACK
+        # Reaching here means, everything looks good
+        self._logger.debug("Current Laserhead Max temp:{}".format( current_laserhead_properties["max_temperature"]))
+        return current_laserhead_properties["max_temperature"]
+
+    def _load_current_laserhead_properties(self):
+        """
+        Loads the current detected laser head related properties and return them
+        Returns:
+            Current laser head properties (dict), (None) otherwise
```
Comment (JFSolution) [inline octoprint_mrbeam/iobeam/laserhead_handler.py]: same here, we need a space between description and returns/args

File: octoprint_mrbeam/iobeam/laserhead_handler.py [hunk 3]
```diff
@@ -407,3 +409,65 @@ def _write_laser_heads_file(self, laser_heads_file=None):
...
+
+        """
+        # 1. get the ID of the current laser head
+        laserhead_id = self.get_current_used_lh_model_id()
+
+        # 2. Load the corresponding yaml file and return it's content
+        lh_properties_file_path = os.path.join("/home/pi/oprint/lib/python2.7/site-packages/octoprint_mrbeam",
```
Comment (JFSolution) [inline octoprint_mrbeam/iobeam/laserhead_handler.py]: I would not hardcode the path, you could use the path of the package itselfe `plugin._basefolder`

**Review by JFSolution — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/iobeam/laserhead_handler.py (hunk 3 above)
Comment (ahmed-mrbeam) [reply octoprint_mrbeam/iobeam/laserhead_handler.py]: Thank you, was looking for this one :). Done

File: octoprint_mrbeam/iobeam/laserhead_handler.py (hunk 2 above)
Comment (ahmed-mrbeam) [reply octoprint_mrbeam/iobeam/laserhead_handler.py]: Done, Thanks

File: octoprint_mrbeam/iobeam/laserhead_handler.py (hunk 1 above)
Comment (ahmed-mrbeam) [reply octoprint_mrbeam/iobeam/laserhead_handler.py]: Done, Thanks

**Review by JFSolution — APPROVED** [review]

File: octoprint_mrbeam/iobeam/laserhead_handler.py [hunk 4]
```diff
@@ -407,3 +409,67 @@ def _write_laser_heads_file(self, laser_heads_file=None):
...
         except IOError as e:
             self._logger.error("Can't open file: {} , Due to error: {}: ".format(laser_heads_file, e))
+
+    @property
+    def current_laserhead_max_temperature(self):
+        """
+        Return the current laser head max temperature
+
+        Returns:
+            Laser head max temp (float)
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/iobeam/laserhead_handler.py]: Just for a matter of consistency with other comments, I'd rephrase this like other comments in this file:
`float: Laser head max temp`

File: octoprint_mrbeam/iobeam/laserhead_handler.py [hunk 5]
```diff
@@ -407,3 +409,67 @@ def _write_laser_heads_file(self, laser_heads_file=None):
...
+        # Reaching here means, everything looks good
+        self._logger.debug("Current Laserhead Max temp:{}".format( current_laserhead_properties["max_temperature"]))
+        return current_laserhead_properties["max_temperature"]
+
+    def _load_current_laserhead_properties(self):
+        """
+        Loads the current detected laser head related properties and return them
+
+        Returns:
+            Current laser head properties (dict), (None) otherwise
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/iobeam/laserhead_handler.py]: `dict: current laser head properties, None: otherwise`

File: octoprint_mrbeam/iobeam/laserhead_handler.py [hunk 6]
```diff
@@ -407,3 +409,67 @@ def _write_laser_heads_file(self, laser_heads_file=None):
...
+
+        """
+        # 1. get the ID of the current laser head
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/iobeam/laserhead_handler.py:446]: I like this ordered list style of comments!

**Review by khaledsherkawi — APPROVED** [review]

File: octoprint_mrbeam/iobeam/laserhead_handler.py (hunk 5 above)
Comment (ahmed-mrbeam) [reply octoprint_mrbeam/iobeam/laserhead_handler.py]: Thanks, Addressed

File: octoprint_mrbeam/iobeam/laserhead_handler.py (hunk 4 above)
Comment (ahmed-mrbeam) [reply octoprint_mrbeam/iobeam/laserhead_handler.py]: Thanks, Done
